tests/CppWinRTTests.cpp

```diff
@@ -42,8 +42,52 @@ void CheckMapVector(std::vector<winrt::Windows::Foundation::Collections::IKeyVal
     }
 }
 
+struct vector_like
+{
+    uint32_t Size() const { return 100; }
+    int GetAt(uint32_t) const { return 15; }
+    uint32_t GetMany(uint32_t start, winrt::array_view<int> items) const
+    { 
+        if (start > 0) throw winrt::hresult_out_of_bounds(); 
+        std::fill_n(items.begin(), (std::min)(items.size(), Size()), GetAt(0));
+        return Size();
+    }
+};
+
+struct iterator_like
+{
+    static const uint32_t total = 20;
+    mutable uint32_t remaining = total;
+    int Current() const { return 3; }
+    uint32_t GetMany(winrt::array_view<int> items) const
+    {
+        auto to_copy = (std::min)(items.size(), remaining);
+        std::fill_n(items.begin(), to_copy, Current());
+        remaining -= to_copy;
+        return to_copy;
+    }
+};
+
+
+struct iterable_like
+{
+    auto First() const { return iterator_like{}; }
+};
+
+struct unstable_vector : winrt::implements<unstable_vector, winrt::Windows::Foundation::Collections::IVectorView<int>>
+{
+    auto Size() { return 4; }
+    int GetAt(uint32_t) { return 7; }
+    uint32_t GetMany(uint32_t, winrt::array_view<int> items) { 

```

Comment:
Size, GetAt, GetMany and IndexOf aren't `const` methods?

Because their inherited and not const in the parent (unstable_vector?)

Reply:
This is the implementation of IVector for C++/WinRT. _Inside_ the implementation, C++/WinRT calls the non-const version of the type.  The _projection_ of IVector<T> is all const methods.